addons/website_sale/models/sale_order.py

```diff
@@ -23,6 +23,16 @@ class SaleOrder(models.Model):
     payment_acquirer_id = fields.Many2one('payment.acquirer', string='Payment Acquirer', copy=False)
     payment_tx_id = fields.Many2one('payment.transaction', string='Last Transaction', copy=False)
     only_services = fields.Boolean(compute='_compute_cart_info', string='Only Services')
+    can_directly_mark_as_paid = fields.Boolean(compute='_compute_can_directly_mark_as_paid',
```

Comment:
Probably merge https://github.com/odoo/odoo/pull/17067/commits/aec41a40d3b5d33b2639d6bba554fe1f807e59f5 in this one.